[WeideTeo] iP

WeideTeo opened the pull request:
# Duke ChatBot  

> Arnold Schwarzenegger  [Bodybuilding Motivation](https://www.youtube.com/watch?v=backYFUEX-Y&ab_channel=%E9%AD%94%E9%AC%BC%E9%98%BF%E8%AB%BE-Arnold)

![This is an image](https://imageio.forbes.com/blogs-images/rainerzitelmann/files/2019/05/GettyImages-481406443-1200x1785.jpg?format=jpg&width=1200) 

- Good GUI 
- Good to Keep Track your Schedule and prepare for more ~~gym~~ ***GAINS!!***
- Buy now

How to use:  
1. Run the Launcher 
2. Type your tasks in a specific order
3. Done !! 😉 

***DOWNLOAD NOW AND GET FREE PROTEIN POWDER !!***

Features:
- [x] Managing tasks  
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

```Java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
} 
```

Commit:
Add Gradle support

WeideTeo committed:
Level 1

WeideTeo committed:
Level 2

WeideTeo committed:
Level 3

WeideTeo committed:
Level 3

WeideTeo committed:
Level 4

WeideTeo committed:
Level 5

WeideTeo committed:
Automated Text UI Testing

WeideTeo committed:
Level 6

WeideTeo committed:
A-TextUiTesting

WeideTeo committed:
Level-7

WeideTeo committed:
Duke txt sample file

WeideTeo committed:
Level-8

WeideTeo committed:
Level-7

WeideTeo committed:
Merge remote-tracking branch 'origin/branch-Level-8' into branch-Level-8

WeideTeo committed:
Actual Level 8: Revert "Level-7"

This reverts commit f355e774a6052579be78fc86d7e08f30144918d9.

WeideTeo committed:
Revert "Level-7"

This reverts commit f355e774a6052579be78fc86d7e08f30144918d9.

WeideTeo committed:
Merge remote-tracking branch 'origin/branch-Level-8' into branch-Level-8

WeideTeo committed:
Level-8

WeideTeo committed:
A-MoreOOP

WeideTeo committed:
A-Packages

WeideTeo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

WeideTeo committed:
A-Gradle

WeideTeo committed:
A-JUnit

WeideTeo committed:
A-JUnit

WeideTeo committed:
A-Jar

WeideTeo committed:
A-Docs

WeideTeo committed:
Updates on Java Docs

WeideTeo committed:
A-CodingStandard

WeideTeo committed:
A-CodingStandard

JrmCkh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,63 @@
+        Scanner scanner = new Scanner(System.in);
+
+            while (true) {
+                try {
```
I think that there is an indentation error starting from your while-clause. The indentation should start at the same line as your scanner. 🤔

JrmCkh commented on src/main/java/duke/DateStringConverter.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    /***
+     *  A method to convert a String type to a LocalTime type
```
I think comments should use 2 * instead of 3 * ? 🤔 There seems to be an additional spacing before your first comment line as well.

JrmCkh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+            return icon + super.toString();
+        }
+    }
+}
```
I think all your javadoc comments seems to start with 3 * instead of 2 *. You might want to change to 2 to align to java coding standard 🤔

JrmCkh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+            return icon + super.toString() + "(by: " + date + ")";
+        } else {
+            return icon + super.toString();
+        }
```
Any reason why is there a need for 3 return statements? 🤔 Especially since to initialize a deadline task, your constructor would take in both date and time as an argument.

JrmCkh commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,39 @@
+        super(details);
+        this.from = from.replace("from", "");
+        this.to = to.replace("to", "");
+    }
```
I think for quite a few of your methods' Javadoc comment you are missing a descriptive header comment.

JrmCkh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,83 @@
+        } else {
+            System.out.println("OOPS!! I'm sorry, but I don't know what that means :-(");
+            return false;
+        }
```
Perhaps you would like to consider using a switch statement instead of a if-else statement? 🤔 It could possibly make it neater overall.

JrmCkh commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,61 @@
+ *  The superclass of all types of Tasks created by the user
+ */
+public class Task {
+    private String details;
```
Perhaps you would like to make Task an abstract class? Especially since you would not need to instantiate this class directly. 🤔

JrmCkh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,236 @@
+                    "Day of the week (E.g. Monday, Tuesday, Wednesday).\n" +
+                    "If time is provided, please key in this format 1200 after entering the date");
+        }
+    }
```
Perhaps you would like to put the functionality of computing deadline, event, and todo into parser class? 🤔 It would make this class neater.

JrmCkh reviewed (commented):
Great code overall! 😄

lxz333 commented on src/main/java/duke/DateStringConverter.java:
```diff
@@ -0,0 +1,43 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/***
```
It would be better if you could change the opening into "/**" so that it will fit our javaDoc form.

lxz333 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,39 @@
+ * Event class type, a subclass of Task
+ */
+public class Event extends Task {
+    final String ICON = "[E]";
```
Maybe can consider adding a single line javadoc for class members.

lxz333 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,61 @@
+    /***
+     * A method to change status to true indicating task is not done
+     */
+    public void unmarked() {
```
Perhaps can consider using verb "unmark" instead of "unmarked".

lxz333 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,236 @@
+     */
+    public void find(String key, ArrayList<Task> taskArrayList) {
+        boolean noResult = true;
+            for(int i=0; i < taskArrayList.size();i++) {
```
It would be better to have a space between "for" and "(" and also next to "=".

lxz333 reviewed (commented):
Overall the code looks good except for some minor coding standard related issues:)

WeideTeo committed:
A-CheckStyle

WeideTeo committed:
Level-10
Implement basic UI and integration.

WeideTeo committed:
Update README.md

WeideTeo committed:
Revert "Update README.md"

This reverts commit 78e5b7db317ed4d9b838790193057b04aa767e90.

WeideTeo committed:
Implement Assertions,

Mainly is for TaskList methods:
Delete, Mark, Unmark, where there might be indexOutOfBound. Hence, assertions are included to prevent such occurence.

WeideTeo committed:
Implement Assertions,

Mainly is for TaskList methods:
Delete, Mark, Unmark, where there might be indexOutOfBound. Hence, assertions are included to prevent such occurence.

WeideTeo committed:
Implement Assertions,

Mainly is for TaskList methods:
Delete, Mark, Unmark, where there might be indexOutOfBound. Hence, assertions are included to prevent such occurence.

WeideTeo committed:
Implement Assertions,

Mainly is for TaskList methods:
Delete, Mark, Unmark, where there might be indexOutOfBound. Hence, assertions are included to prevent such occurence.

WeideTeo committed:
Amendment done to checkstyle quality such as arrow head code, insufficient encapsulation, confusing naming.

WeideTeo committed:
Revert "Implement Assertions,"

This reverts commit cd70a722d9a1dc833862ca7b9fa9f77e939634c7.

WeideTeo committed:
Amendment done for assertion due to merge conflicts

WeideTeo committed:
Amendment done for coding quality due to merge conflicts

WeideTeo committed:
C-MassOps

Done previously, delete all tasks in arraylist and also delete all the contents in the txt file.

Refer to class Storage.deleteAll function

WeideTeo committed:
B-ViewSchedules

Done previously, allow you to type in command:

by Sunday

It will show you the deadline tasks before Sunday. Only deadline not event.

WeideTeo committed:
Fix Bugs for Deadline, Mark and Unmark Functions

WeideTeo committed:
A-Assertion changes to gradlew file

WeideTeo committed:
Bug solve for delete function

WeideTeo committed:
Bug solve for mkdir

WeideTeo committed:
Update README.md

WeideTeo committed:
Storage Class: saveData and loadData method

Change !userData.exists() to userData.exists == false for improved readibility

A case insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

In addition, the previous commit was edition was to make changes to the relative path, as the path folder was created with incorrect name

Let's,
* Push the new version

WeideTeo committed:
Storage Class: saveData and loadData method

Change !userData.exists() to userData.exists == false for improved readibility

A case insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

In addition, the previous commit was edition was to make changes to the relative path, as the path folder was created with incorrect name

Let's,
* Push the new version

WeideTeo committed:
Merge remote-tracking branch 'origin/master'

WeideTeo committed:
A-Release
